specs/core/0_fork-choice.md

```diff
@@ -136,17 +136,69 @@ def get_latest_attesting_balance(store: Store, root: Root) -> Gwei:
     ))
 ```
 
+#### `filter_block_tree`
+
+```python
+def filter_block_tree(store: Store, block_root: Root, blocks: Dict[Root, BeaconBlock]) -> bool:
+    block = store.blocks[block_root]
+    children = [
+        root for root in store.blocks.keys()
+        if store.blocks[root].parent_root == block_root
+    ]
+
+    # If any children branches contain expected finalized/justified checkpoints,
+    # add to filtered block-tree and signal viability to parent.
+    if any(children):
+        if True in [filter_block_tree(store, child, blocks) for child in children]:
```

Comment:
```suggestion
        if any(filter_block_tree(store, child, blocks) for child in children):
```

Reply:
Tricky. This actually breaks things. If you pass an iterator into `any`, then the iterator stops running upon first item that is `True`. Because I am relying upon the side effect of `filter_block_tree` run on all children, then this can truncate valid branches

Example from python console
```
>>> def test(i):
...     print(i)
...     return i > 4
... 
>>> any(test(i) for i in range(10))
0
1
2
3
4
5
True
>>> 
```

Reply:
But if you put the iterator inside a list def, it evaluates all

```
>>> any([test(i) for i in range(10)])
0
1
2
3
4
5
6
7
8
9
True
```

Reply:
Well, it's probably better to be more explicit about that side-effect then. Implementers looking to optimize things even more than this, and are likely to make the same mistake.

Reply:
To be more explicit, maybe split into two lines?

```suggestion
    filter_block_tree_result = [filter_block_tree(store, child, blocks) for child in children]
    if any(filter_block_tree_result):
```